src/reference.rs

```diff
@@ -179,6 +179,19 @@ impl<'repo> Reference<'repo> {
         }
     }
 
+    /// Conditionally create a new reference with the same name as the given
+    /// reference but a different OID target. The reference must be a direct
+    /// reference, otherwise this will fail.
+    pub fn set_target(&mut self, id: Oid, msg: &str) -> Result<Reference<'repo>, Error> {
```

Comment:
So when I pass in `0 as *const libc::c_char`  for msg (to the raw call), everything seems to work (in at least one particular case).  The reflog records an empty message.

Do you  want  me to make `msg` an `Option`?  I could do that but then we might want  to  modify the  existing `reference.rename()` function, which also takes a  msg argument, to be the same.